instrumentation/openai-client-instrumentation/instrumentation-0.13.0/src/main/java/co/elastic/otel/openai/v0_13_0/ApiAdapterImpl.java

```diff
@@ -0,0 +1,117 @@
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package co.elastic.otel.openai.v0_13_0;
+
+import co.elastic.otel.openai.wrappers.ApiAdapter;
+import com.openai.models.ChatCompletionAssistantMessageParam;
+import com.openai.models.ChatCompletionContentPart;
+import com.openai.models.ChatCompletionCreateParams;
+import com.openai.models.ChatCompletionMessageParam;
+import com.openai.models.ChatCompletionSystemMessageParam;
+import com.openai.models.ChatCompletionToolMessageParam;
+import com.openai.models.ChatCompletionUserMessageParam;
+
+public class ApiAdapterImpl extends ApiAdapter {
+
+  public static void init() {
+    ApiAdapter.init(ApiAdapterImpl::new);
+  }
+
+  @Override
+  public Object extractConcreteCompletionMessageParam(ChatCompletionMessageParam base) {
+    if (base.isChatCompletionSystemMessageParam()) {
+      return base.asChatCompletionSystemMessageParam();
+    }
+    if (base.isChatCompletionUserMessageParam()) {
+      return base.asChatCompletionUserMessageParam();
+    }
+    if (base.isChatCompletionAssistantMessageParam()) {
+      return base.asChatCompletionAssistantMessageParam();
+    }
+    if (base.isChatCompletionToolMessageParam()) {
+      return base.asChatCompletionToolMessageParam();
+    }
+    throw new IllegalStateException("Unhandled message param type: " + base);
+  }
+
+  @Override
+  public String extractText(ChatCompletionContentPart part) {
+    if (part.isChatCompletionContentPartText()) {
+      return part.asChatCompletionContentPartText().text();
+    }
+    return null;
+  }
+
+  @Override
+  public String extractType(ChatCompletionCreateParams.ResponseFormat val) {
+    if (val.isResponseFormatText()) {
+      return val.asResponseFormatText()._type().toString();
+    } else if (val.isResponseFormatJsonObject()) {
+      return val.asResponseFormatJsonObject()._type().toString();
+    } else if (val.isResponseFormatJsonSchema()) {
+      return val.asResponseFormatJsonSchema()._type().toString();
+    }
+    return "";
+  }
+
+  @Override
+  public String extractTextOrRefusal(
+      ChatCompletionAssistantMessageParam.Content.ChatCompletionRequestAssistantMessageContentPart
+          part) {
+    if (part.isChatCompletionContentPartText()) {
+      return part.asChatCompletionContentPartText().text();
+    }
+    if (part.isChatCompletionContentPartRefusal()) {
+      return part.asChatCompletionContentPartRefusal().refusal();
+    }
+    return null;
+  }
+
+  @Override
+  public String asText(ChatCompletionUserMessageParam.Content content) {
+    if (content.isTextContent()) {
+      return content.asTextContent();
+    }
+    return "";
```

Comment:
[minor] maybe use a static method to avoid duplication.

Reply:
I switched to a ternary operator to make thinks more concise now. Other than that, I don't think we can extract anything here, all the `Content` types across the methods are differen.